### PR TITLE
Add preset tests

### PR DESCRIPTION
**Pull request by hwmarkcheng**
## Checklist
Adding Preset Integration and unit tests for the Preset ingestion method.

`Test setup:`
- [https://datahubproject.io/docs/metadata-ingestion/developing/#:~:text=all dev requirements.-,pip install -e '.[dev]',-%23 For running integration](https://datahubproject.io/docs/metadata-ingestion/developing/#:~:text=all%20dev%20requirements.-,pip%20install%20-e%20%27.%5Bdev%5D%27,-%23%20For%20running%20integration)
- linting: ./gradlew :metadata-ingestion:lintFix
- Need to first install dependencies:
    ```sql
    cd metadata-ingestion
    ../gradlew :metadata-ingestion:installDev
    source venv/bin/activate
    datahub version  # should print "DataHub CLI version: unavailable (installed in develop mode)"
    ```
    
- For pytest:
    - `run pytest tests/integration/preset/test_preset.py`
    - `run pytest tests/unit/test_preset_source.py`

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by hwmarkcheng — COMMENTED** [review]
#TODO still need to add mock request for `test_response = self.session.get(f"{self.config.connect_uri}/version")`

File: metadata-ingestion/src/datahub/ingestion/source/preset.py [hunk 1]
```diff
@@ -33,11 +33,11 @@
 logger = logging.getLogger(__name__)
 class PresetConfig(StatefulIngestionConfigBase, ConfigModel):
     manager_uri: str = Field(
-        default="https://api.app.preset.io/", description="Preset.io API URL"
+        default="https://api.app.preset.io", description="Preset.io API URL"
     )
-    connect_uri: str = Field(default=None, description="Preset workspace URL.")
+    connect_uri: str = Field(default='https://api.app.preset.io', description="Preset workspace URL.")
```
Comment (llance) [inline metadata-ingestion/src/datahub/ingestion/source/preset.py]: connect_uri doesn't default to 'https://api.app.preset.io' only manager_uri does. 

see: https://github.com/wealthsimple/datahub-infrastructure/commit/59940a21b3e0f8a02be1cfa2c743a2c555faa8a5#diff-272ee977e0f6e17fbc78e356121d288ab1a67a97605c3f51537028804671187fR7

let's leave it to None for now because it varies between workspace to workspace, and company to company.

File: metadata-ingestion/src/datahub/ingestion/source/preset.py [hunk 2]
```diff
@@ -33,11 +33,11 @@
...
     display_uri: Optional[str] = Field(
-        default=None,
+        default='https://api.app.preset.io',
```
Comment (llance) [inline metadata-ingestion/src/datahub/ingestion/source/preset.py]: I'm not sure exactly how the underlying mechanism work for display_uri, but I don't think we should default this to equal the same value as connect_uri. Especially when display_uri's description is 
"optional URL to use in links (if `connect_uri` is only for ingestion)" 
let's leave it as None for now to follow same convention as superset

reference: https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/superset.py#L96

File: metadata-ingestion/src/datahub/ingestion/source/superset.py [hunk 3]
```diff
@@ -108,7 +108,7 @@ class SupersetConfig(
     api_key: Optional[str] = Field(default=None, description="Preset.io API key.")
     api_secret: Optional[str] = Field(default=None, description="Preset.io API secret.")
     manager_uri: str = Field(
-        default="https://api.app.preset.io/", description="Preset.io API URL"
```
Comment (llance) [inline metadata-ingestion/src/datahub/ingestion/source/superset.py:111]: I think I need to do some refactoring. I don't think api_key, api_secret and manager_uri should belong to SupersetConfig class since it is only relevant to PresetConfig. Ill put up a separate PR to remove those fields from SupersetConfig and you can rebase against master after it gets merged.

File: metadata-ingestion/src/datahub/ingestion/source/preset.py (hunk 1 above)
Comment (hwmarkcheng) [reply metadata-ingestion/src/datahub/ingestion/source/preset.py]: Makes sense, I'll edit the unit test to reflect this

File: metadata-ingestion/src/datahub/ingestion/source/superset.py (hunk 3 above)
Comment (hwmarkcheng) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py:111]: Yeah, I think the only preset portion this is used might be the authentication endpoint? 
Preset API is split to preset specific/superset:

https://api-docs.preset.io/#cdf2f4d7-1842-4a3e-bd15-a747295704ee

File: metadata-ingestion/src/datahub/ingestion/source/preset.py (hunk 2 above)
Comment (hwmarkcheng) [reply metadata-ingestion/src/datahub/ingestion/source/preset.py]: Hm right, wasn't too sure of this either, good to leave as none for now
